geomesa-arrow/geomesa-arrow-jts/src/main/java/org/locationtech/geomesa/arrow/jts/impl/AbstractLineStringVector.java

```diff
@@ -81,8 +86,14 @@ public LineString get(int index) {
       final int offsetEnd = vector.getOffsetBuffer().getInt((index + 1) * ListVector.OFFSET_WIDTH);
       final Coordinate[] coordinates = new Coordinate[offsetEnd - offsetStart];
       for (int i = 0; i < coordinates.length; i++) {
-        final double y = readOrdinal((offsetStart + i) * 2);
-        final double x = readOrdinal((offsetStart + i) * 2 + 1);
+        final double y, x;
+        if (getFlipAxisOrder()) {
+          y = readOrdinal(x(offsetStart + i));
```

Comment:
i think i'd get rid of the `y()` and `x()` methods and just keep the math here like it was, since it's an extra method call (which might be inlined but might not be), and b/c sometimes we're setting y to x() which seems kind of confusing.

Reply:
well, absent some benchmarks that show performance changes one way or another, I would like to avoid extra method calls. this code gets called potentially millions of times in a given request, so I don't think it's premature to optimize it (or at least try to avoid regression). can we accomplish the same thing with better javadoc comments? I think knowing that x/y are interleaved in the single arrow vector makes the offset calculations fairly transparent.

Reply:
- Does geomesa have any kind of benchmark baseline methodology (performance suite) and numbers to compare to?

- If you insist I can revert the `y()` and `x()` functions but again I think for a request with a monster sized geometry with millions of points to process this kind of optimization is really derived by the JVM HotSpot at runtime and the run time ceiling will most likely not be affacted by the extra dispatches to `y()` and `x()` but by I/O related tasks.

Reply:
you're probably right that it won't matter, but I'm cautious about changes - i'd prefer to err on the side of performance for low-level functions like this. we did have some JMH-based benchmarks that we took out b/c we were no longer using them, but if you want to resurrect them it shouldn't be too difficult: https://github.com/locationtech/geomesa/blob/3.5.x/benchmarks/src/main/scala/org/locationtech/geomesa/benchmarks/JsonPathBenchmarks.scala